.travis.yml

```diff
@@ -3,11 +3,48 @@
 
 ---
 language: python
+
 python:
-    - "2.6"
-    - "2.7"
-install:
-    - pip install pycodestyle
+  - 2.6
+  - 2.7
+env:
+  - RUN_COVERAGE=false
+  - RUN_COVERAGE=true
+
+matrix:
+  fast_finish: true
+  exclude:
+    - python: 2.6
+      env: RUN_COVERAGE=true
+    - python: 2.7
+      env: RUN_COVERAGE=false
+
+before_install:
+  # Coverage dependencies
+  - pip install coverage pytest-cov
+  # Code style dependencies
+  - pip install pycodestyle
 script:
-    - python -m isodatetime.tests
-    - pycodestyle -v isodatetime
+  - |
+    export COVERAGE_PROCESS_START="${TRAVIS_BUILD_DIR}/.coveragerc"
+    # NOTE: the way to handle long-living commands in the documentation is
+    # travis_wait 40 coverage run -m isodatetime.tests
+    # but it does not produce any output, which can be frustrating in case of errors.
+    # So here we are using the approach documented in the issue
+    # https://github.com/travis-ci/travis-ci/issues/4190.
+    coverage run -m isodatetime.tests &
+    minutes=0
+    limit=40
+    while kill -0 $! >/dev/null 2>&1; do
+      echo -n -e " \b" # never leave evidences!
+      if [ $minutes == $limit ]; then
+        break;
+      fi
+      minutes=$((minutes+1))
+      sleep 60
+    done
```

Comment:
Can we just use `timeout 40m coverage ...` here?

Reply:
What sorcery is it?! I never heard about `timeout`. First I googled for Travis and timeout, as I thought it would be something from Travis... but then realized it could be actually one of the many great GNU core utils that I am still not familiar with! http://www.gnu.org/software/coreutils/manual/coreutils.html#timeout-invocation #TodayILearned

Will finish reading the man page, and play with timeout in a bit. If it's not too late - and it works - will update it to use `timeout` instead in the next minutes. Otherwise it should be ready tomorrow :-) Thanks @matthewrmshin !!!

Reply:
Really simple, didn't spend nearly as close as I thought I would need learning the basics of this tool. Will definitely remember to use it when I have a similar use case. Thanks a lot @matthewrmshin 

Left a [note](https://github.com/travis-ci/travis-ci/issues/4190#issuecomment-426972986) in the Travis-CI issue where I found the previous solution. Hopefully that will help others that like me didn't know `timeout`.

Pushed a new commit with the fix, and also rebased the branch.

Reply:
Ooohh, true. I remember when I initially had the problem it was caused by Travis aborting the build. Not a problem, I ended up learning about timeout which will be handy in the future for sure. Reverting it :-)

Reply:
Just background a process which calls `echo -n -e " \b"` every now an again?

Reply:
Yup. Travis aborts the build if there's no output for more than 10 minutes. 

Coverage added more time to a test that took a while already, resulting in no output for a little over 10 minutes.

Travis provides `travis_wait` which works like `timeout`, preventing the job of being aborted.

However, `travis_wait` does not output anything. Even though the tests emit some useful info - especially when it fails! - so another developer came up with this poor-man's `timeout` variant in a github issue in travisci repo (link in the comment).

And I shamelessly copied the code here, changing only the timeout to 40 minutes (build took between 22-30 minutes IIRC).